Apache-Arrow-Flight-Tester/common.py

```diff
@@ -0,0 +1,108 @@
+import pprint
+from typing import Literal
+
+import pyarrow
+from pyarrow import flight, Schema
+from pyarrow._flight import FlightInfo, FlightClient, ActionType, Result, Ticket
+
+
+# PyArrow Functions.
+def list_flights(flight_client: FlightClient) -> list[FlightInfo]:
+    response = flight_client.list_flights()
+
+    return list(response)
+
+
+def get_schema(flight_client: FlightClient, table_name: str) -> Schema:
+    upload_descriptor = flight.FlightDescriptor.for_path(table_name)
+    response = flight_client.get_schema(upload_descriptor)
+
+    return response.schema
+
+
+def do_get(flight_client: FlightClient, ticket: Ticket) -> None:
+    response = flight_client.do_get(ticket)
+
+    for batch in response:
+        pprint.pprint(batch.data.to_pydict())
+
+
+def do_put(flight_client: FlightClient, table_name: str, record_batch: pyarrow.RecordBatch) -> None:
+    upload_descriptor = flight.FlightDescriptor.for_path(table_name)
+    writer, _ = flight_client.do_put(upload_descriptor, record_batch.schema)
+
+    writer.write(record_batch)
+    writer.close()
+
+
+def do_action(flight_client: FlightClient, action_type: str, action_body: bytes) -> list[Result]:
+    action = flight.Action(action_type, action_body)
+    response = flight_client.do_action(action)
+
+    return list(response)
+
+
+def list_actions(flight_client: FlightClient) -> list[ActionType]:
+    response = flight_client.list_actions()
+
+    return list(response)
+
+
+# Helper functions.
+def list_table_names(flight_client: FlightClient) -> list[str]:
+    flights = list_flights(flight_client)
+
+    return [table_name.decode("utf-8") for table_name in flights[0].descriptor.path]
+
+
+def update_object_store(flight_client: flight.FlightClient, object_store_type: Literal["s3", "azureblobstorage"],
```

Comment:
Maybe replace `object_store_type` and `arguments` with a single parameter, e.g., named `object_store_data`, that accepts either an object of a class that only contains the necessary data for `s3` as instance variables or an object of a class that contains only the necessary data for `azureblobstorage` as instance variables. Maybe make the two classes derive from a super class or a protocol that simply provides a method that returns the arguments so the loop becomes `object_store_data.arguments()` and the type of `object_store_data` can be specified as the super class or protocols. This should allow the Python type system to make it clear what information is needed instead of a comment.

Reply:
Good idea, fixed.